### ISSUE TITLE
MemoryError in Hyperopt

### ISSUE THREAD

**Issue by MicahMosier** [freqtrade/freqtrade#2296]
## Step 1: Yes

If you have discovered a bug in the bot, please [search our issue tracker](https://github.com/freqtrade/freqtrade/issues?q=is%3Aissue). 
If it hasn't been reported, please create a new issue.

## Step 2: Describe your environment

  * Operating system: Debian Buster
  * Python Version: 3.7 (`python -V`)
  * CCXT version: ccxt==1.18.1159 (`pip freeze | grep ccxt`)
  * Branch: Master
  * Last Commit ID:  ?  (`git log --format="%H" -n 1`)
  
## Step 3: Describe the problem:

Hyperopt fails out with a "MemoryError". It may be because I am using a raspberry pi 4 (4GB of RAM version.) But I feel like it should still work. It looks like it's running out of usable memory. Is there a way to store old results on the microSD card?

### Steps to reproduce:

  1. Start hyperopt
  2. Wait for failure
    
### Observed Results:

  Hyperopt experiences memory failure when I run over 100 epochs.

### Relevant code exceptions or logs:

  ```
  2019-09-23 08:28:14,043 - freqtrade.optimize.hyperopt - INFO - Effective number of parallel workers used: 4
/home/freq/freqtrade/.env/lib/python3.7/site-packages/sklearn/externals/joblib/__init__.py:15: DeprecationWarning: sklearn.externals.joblib is deprecated in 0.21 and will be removed in 0.23. Please import this functionality directly from joblib, which can be installed with: pip install joblib. If this warning is raised when loading pickled models, you may need to re-serialize those models with scikit-learn 0.21+.
  warnings.warn(msg, category=DeprecationWarning)
/home/freq/freqtrade/.env/lib/python3.7/site-packages/sklearn/externals/joblib/__init__.py:15: DeprecationWarning: sklearn.externals.joblib is deprecated in 0.21 and will be removed in 0.23. Please import this functionality directly from joblib, which can be installed with: pip install joblib. If this warning is raised when loading pickled models, you may need to re-serialize those models with scikit-learn 0.21+.
  warnings.warn(msg, category=DeprecationWarning)
/home/freq/freqtrade/.env/lib/python3.7/site-packages/sklearn/externals/joblib/__init__.py:15: DeprecationWarning: sklearn.externals.joblib is deprecated in 0.21 and will be removed in 0.23. Please import this functionality directly from joblib, which can be installed with: pip install joblib. If this warning is raised when loading pickled models, you may need to re-serialize those models with scikit-learn 0.21+.
  warnings.warn(msg, category=DeprecationWarning)
/home/freq/freqtrade/.env/lib/python3.7/site-packages/sklearn/externals/joblib/__init__.py:15: DeprecationWarning: sklearn.externals.joblib is deprecated in 0.21 and will be removed in 0.23. Please import this functionality directly from joblib, which can be installed with: pip install joblib. If this warning is raised when loading pickled models, you may need to re-serialize those models with scikit-learn 0.21+.
  warnings.warn(msg, category=DeprecationWarning)

*    1/1000:    212 trades. Avg profit -0.73%. Total profit -0.07729898 USD (-154.21Σ%). Avg duration 1195.5 mins. Objective: 2.71711

*    2/1000:    131 trades. Avg profit -0.31%. Total profit -0.02029972 USD ( -40.50Σ%). Avg duration 220.5 mins. Objective: 2.25252
....................................................................................................................................................................................................................
   215/1000:     16 trades. Avg profit  0.11%. Total profit  0.00084902 USD (   1.69Σ%). Avg duration 23194.1 mins. Objective: 2.24875
.................................
   249/1000:     15 trades. Avg profit  0.18%. Total profit  0.00138045 USD (   2.75Σ%). Avg duration 24747.7 mins. Objective: 2.24548
.....................................................
   303/1000:     46 trades. Avg profit -0.78%. Total profit -0.01803470 USD ( -35.98Σ%). Avg duration 155.3 mins. Objective: 2.17333
.
   305/1000:     32 trades. Avg profit -0.43%. Total profit -0.00689702 USD ( -13.76Σ%). Avg duration 198.8 mins. Objective: 2.16094
........
   314/1000:     37 trades. Avg profit -0.85%. Total profit -0.01583057 USD ( -31.58Σ%). Avg duration 153.1 mins. Objective: 2.15814
................
   331/1000:     31 trades. Avg profit -0.83%. Total profit -0.01292363 USD ( -25.78Σ%). Avg duration 162.6 mins. Objective: 2.15306

   332/1000:     29 trades. Avg profit -0.84%. Total profit -0.01214960 USD ( -24.24Σ%). Avg duration 165.2 mins. Objective: 2.15191

   333/1000:     32 trades. Avg profit -0.78%. Total profit -0.01251439 USD ( -24.97Σ%). Avg duration 149.7 mins. Objective: 2.13288
..............
   348/1000:     36 trades. Avg profit -0.67%. Total profit -0.01211746 USD ( -24.17Σ%). Avg duration 150.1 mins. Objective: 2.12976
.............................................................
   410/1000:     44 trades. Avg profit -0.64%. Total profit -0.01404700 USD ( -28.02Σ%). Avg duration 126.5 mins. Objective: 2.10888
..............................................................................................................................................................................................................................................................................................................................................................................................................2019-09-23 14:22:57,270 - freqtrade - ERROR - Fatal exception!
joblib.externals.loky.process_executor._RemoteTraceback:
'''
Traceback (most recent call last):
  File "/home/freq/freqtrade/.env/lib/python3.7/site-packages/joblib/externals/loky/process_executor.py", line 391, in _process_worker
    call_item = call_queue.get(block=True, timeout=timeout)
  File "/usr/lib/python3.7/multiprocessing/queues.py", line 108, in get
    res = self._recv_bytes()
  File "/usr/lib/python3.7/multiprocessing/connection.py", line 216, in recv_bytes
    buf = self._recv_bytes(maxlength)
  File "/usr/lib/python3.7/multiprocessing/connection.py", line 411, in _recv_bytes
    return self._recv(size)
  File "/usr/lib/python3.7/multiprocessing/connection.py", line 386, in _recv
    buf.write(chunk)
MemoryError
'''

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/freq/freqtrade/freqtrade/main.py", line 39, in main
    args['func'](args)
  File "/home/freq/freqtrade/freqtrade/optimize/__init__.py", line 74, in start_hyperopt
    hyperopt.start()
  File "/home/freq/freqtrade/freqtrade/optimize/hyperopt.py", line 410, in start
    f_val = self.run_optimizer_parallel(parallel, asked)
  File "/home/freq/freqtrade/freqtrade/optimize/hyperopt.py", line 347, in run_optimizer_parallel
    wrap_non_picklable_objects(self.generate_optimizer))(v) for v in asked)
  File "/home/freq/freqtrade/.env/lib/python3.7/site-packages/joblib/parallel.py", line 934, in __call__
    self.retrieve()
  File "/home/freq/freqtrade/.env/lib/python3.7/site-packages/joblib/parallel.py", line 833, in retrieve
    self._output.extend(job.get(timeout=self.timeout))
  File "/home/freq/freqtrade/.env/lib/python3.7/site-packages/joblib/_parallel_backends.py", line 521, in wrap_future_result
    return future.result(timeout=timeout)
  File "/usr/lib/python3.7/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/usr/lib/python3.7/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed to un-serialize. Please ensure that the arguments of the function are all picklable.

  ```

**Comment by MicahMosier** [comment]
74a0f44230adbefcedc3abddbb30f2722baf4b65 Is the commit ID.

**Comment by hroff-1902** [comment]
Duplicate of #2271,
fixed by #2294

**Comment by hroff-1902** [comment]
Btw, running hyperopt on RPI is not a best idea...

**Comment by MicahMosier** [comment]
Yeah lol I am aware, I am in the process of prototyping an 8 pi4 cluster for trading/hyperopt.

**Comment by xmatthias** [comment]
closed in #2294
